react-web-app/src/utils/signature.test.js

```diff
@@ -0,0 +1,50 @@
+import { getRequestSignature } from './signature';
+
+// This is just a mockup
+const isHmacSha1Base64Hash = (str) => {
+  if (!str) return;
+
+  // Regular expression for Base64 encoding
+  const isValidBase64 = (str) => /^[A-Za-z0-9+/]+={0,2}$/.test(str);
+
+  // Check if the length is 28 characters (without padding) or 40 characters (with padding)
```

Comment:
Are all request signatures exactly this length??

Reply:
This method `isHmacSha1Base64Hash` is just a mockup/simulation example made for the unit test. It doesn't reflect the actual implementation as it is on the front end and doesn't depend on or is affected by the implementation adopted by the merchant on the back end.